/*
 * @begin:license
 *
 * Copyright (c) 2015-2019, Benjamin Niemann <pink@odahoda.de>
 *
 * This program is free software; you can redistribute it and/or modify
 * it under the terms of the GNU General Public License as published by
 * the Free Software Foundation; either version 2 of the License, or
 * (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU General Public License for more details.
 *
 * You should have received a copy of the GNU General Public License along
 * with this program; if not, write to the Free Software Foundation, Inc.,
 * 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
 *
 * @end:license
 */

syntax = "proto2";

package noisicaa.pb;

import "noisicaa/audioproc/public/node_parameters.proto";
import "noisicaa/builtin_nodes/step_sequencer/model.proto";

message StepSequencerSpec {
  required int32 num_steps = 1;
  required bool time_synched = 2;

  message Channel {
    required StepSequencerChannel.Type type = 1;
    repeated bool step_enabled = 2;
    repeated float step_value = 3;
  }
  repeated Channel channels = 3;
}

extend NodeParameters {
  optional StepSequencerSpec step_sequencer_spec = 409000;
}
